server/src/main/java/org/elasticsearch/indices/recovery/RecoveryTarget.java

```diff
@@ -307,6 +307,7 @@ public void finalizeRecovery(final long globalCheckpoint, ActionListener<Void> l
             indexShard.updateGlobalCheckpointOnReplica(globalCheckpoint, "finalizing recovery");
             // Persist the global checkpoint.
             indexShard.sync();
+            indexShard.persistRetentionLeases();
```

Comment:
Do we have a test that shows that we need to do this here? I deleted this line and `:server:unittest` and `:server:integtest` both still passed (once). Is this expected?

Reply:
I pushed 162cf7731198ef603808551b58ce0941d75daf46.